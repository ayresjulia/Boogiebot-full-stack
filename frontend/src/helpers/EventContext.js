import React from "react";

/** Context: provides events object and setter for it throughout app. */

const EventContext = React.createContext();

export default EventContext;
