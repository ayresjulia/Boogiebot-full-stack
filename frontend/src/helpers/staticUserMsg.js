const staticMsg = {
	FORM_TIP: "tip.",
	EDIT_PROFILE_FORM_TIP_DESC:
		"while updating your info, keep in mind, that picking a great playlist is key for any party!",
	EDIT_PROFILE_FORM_CONFIRM_PWD: "Confirm password to make changes",
	FORM_USERNAME: "Username",
	FORM_PWD: "Password",
	FORM_FIRST_NAME: "First Name",
	FORM_LAST_NAME: "Last Name",
	FORM_EMAIL: "Email",
	FORM_PROFILE_URL: "Profile URL",
	EVENT_FORM_TIP_DESC:
		"scoop your ice-cream into a cupcake pan and pop in the freezer for a hassle-free	serving at your party.",
	LOGO_NAME: "BoogieBot",
	NAV_MY_EVENTS: "My Events",
	NAV_INSPIRATION: "Get Inspired",
	NAV_CATERING: "Catering",
	NAV_PROFILE: "Profile",
	NAV_LOG_OUT: "Log Out",
	NAV_LOG_IN: "Log In",
	NAV_SIGN_UP: "Sign Up",
	LOADING: "Loading...",
	CATERING_LOOKUP: "lookup",
	CATERING_FOOTNOTE:
		"check the event you want to save the restaurant to, it will appear in your event details",
	CATERING_INFO: "restaurants nearby with your zip code for catering options.",
	EVENTS_CREATE: "create",
	EVENTS_NEW: "new event.",
	HOME_WELCOME_USER: "Welcome, ",
	NON_USER_MSG: "Please, Log in or Sign up to begin.",
	MY_EVENT_DESC: "Description:",
	MY_EVENT_WHEN: "When:",
	MY_EVENT_TIME: "Time:",
	MY_EVENT_LOCATION: "Location:",
	DEFAULT_EVENT_IMG:
		"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	INSPIRATION_FOOTNOTE:
		"check the event you want to save the picture to, it will appear in your event details",
	INSPIRATION_ADD: "add",
	INSPIRATION_INFO: "pictures to your event moodboard for inspiration.",
	HOME_IMG1:
		"https://images.unsplash.com/photo-1620324446278-be6c7bb3ece5?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG2:
		"https://images.unsplash.com/photo-1527275393322-8ddae8bd5de9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG3:
		"https://images.unsplash.com/photo-1525268771113-32d9e9021a97?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG4:
		"https://images.unsplash.com/photo-1534119238569-839b7389d552?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=800&q=80",
	CONSOLE_EVENTS_ERROR: "ERROR!!! Events could not be loaded",
	CONSOLE_USER_ERROR: "ERROR!!! currentUser.username could not be loaded",
	MY_EVENT_CAT: "catering",
	CATERING_DEFAULT:
		"https://images.unsplash.com/photo-1557962011-4b36a6f5d921?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1350&q=80",
	CATERING_WEB: "website",
	USER_DEFAULT_URL: "https://supersimple.com/wp-content/uploads/peek-a-boo-800-800-200x200.jpg",
	API_ERROR: "There is an error with getting data from API. Please check connection",
	EDIT_EVENT_FORM_TIP_DESC:
		"don't forget about younger audience, set up a cool kids table at your party",
	EDIT_EVENT_FORM_TITLE: "Title",
	EDIT_EVENT_FORM_DESC: "Description",
	EDIT_EVENT_FORM_EVENT_DATE: "Event Date",
	EDIT_EVENT_FORM_EVENT_TIME: "Event Time",
	EDIT_EVENT_FORM_CITY: "City",
	MY_EVENT_RMRK1: "browse pictures in Get Inspired tab to save them here",
	MY_EVENT_RMRK2: "browse restaurants in Catering tab to save them here",
	EVENTS_RMRK:
		"your events will be shown here. click on event to get details or create a new one",
	TARGET_ERR: "targetEvent could not be loaded",
	EVENT_ALERT: "Please choose an event first."
};

export default staticMsg;
