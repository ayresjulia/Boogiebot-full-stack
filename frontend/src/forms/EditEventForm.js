import React, { useState, useContext, useRef, useEffect, useCallback } from "react";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";
import { useParams } from "react-router-dom";
import BoogieBotApi from "../Api";
import { Row, Col } from "react-bootstrap";
import Alert from "../helpers/Alert";
import staticMsg from "../helpers/staticUserMsg";
import states from "../helpers/states";
import countries from "../helpers/countries";
import { v4 as uuid } from "uuid";
import "./EditEventForm.css";
import { useHistory } from "react-router-dom";
import EventContext from "../helpers/EventContext";

/** Form to edit user info in db. */

const EditEventForm = () => {
	const history = useHistory();
	const { events } = useContext(EventContext);
	const [ showAlert, setShowAlert ] = useState(false);
	let { id } = useParams();
	const formDataRef = useRef();

	let targetEvent = events.find((evt) => parseInt(evt.id) === parseInt(id));
	if (!targetEvent) console.error(staticMsg.TARGET_ERR);

	const INITIAL_STATE = {
		title: targetEvent.title,
		description: targetEvent.description,
		eventDate: targetEvent.eventDate,
		eventTime: targetEvent.eventTime,
		city: targetEvent.city,
		state: targetEvent.state,
		country: targetEvent.country,
		imgUrl: targetEvent.imgUrl === staticMsg.DEFAULT_EVENT_IMG ? "" : targetEvent.imgUrl
	};
	const [ formData, setFormData ] = useState(INITIAL_STATE);
	const [ formErrors, setFormErrors ] = useState([]);

	useEffect(() => {
		formDataRef.current = INITIAL_STATE;
		return () => (formDataRef.current = {});
	});

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value
		}));
	};

	/** Submit form and add new event data to db. */

	const handleSubmit = useCallback(
		async (e) => {
			e.preventDefault();
			let eventData = {
				title: formData.title,
				description: formData.description,
				eventDate: formData.eventDate,
				eventTime: formData.eventTime,
				city: formData.city,
				state: formData.state,
				country: formData.country,
				imgUrl: formData.imgUrl
			};

			try {
				await BoogieBotApi.editEvent(id, eventData);
				history.push("/events");
			} catch (err) {
				setFormErrors(err);
				setShowAlert(true);
				console.error(formErrors);
				return;
			}
			setFormData((form) => ({ ...form }));
			setFormErrors([]);
		},
		[ formData, history, formErrors, id ]
	);

	return (
		<div className="Form">
			<div className="Form-left">
				<span className="Form-update">{staticMsg.FORM_TIP}</span>
				<p>{staticMsg.EDIT_EVENT_FORM_TIP_DESC}</p>
			</div>
			<div className="Form-right">
				<div className="font-weight-bold text-center mb-card-title event-title">
					Edit {targetEvent.title}
				</div>
				<Form className="Form-body" onSubmit={handleSubmit}>
					<Row>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="title">{staticMsg.EDIT_EVENT_FORM_TITLE}</Label>
								<Input
									className="Form-input"
									name="title"
									id="title"
									value={formData.title}
									onChange={handleChange}
								/>
							</FormGroup>
						</Col>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="description">
									{staticMsg.EDIT_EVENT_FORM_DESC}
								</Label>
								<Input
									className="Form-input"
									name="description"
									id="description"
									value={formData.description}
									onChange={handleChange}
								/>
							</FormGroup>
						</Col>
					</Row>
					<Row>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="eventDate">
									{staticMsg.EDIT_EVENT_FORM_EVENT_DATE}
								</Label>
								<Input
									className="Form-input"
									type="date"
									name="eventDate"
									id="eventDate"
									value={formData.eventDate}
									onChange={handleChange}
									required
								/>
							</FormGroup>
						</Col>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="eventTime">
									{staticMsg.EDIT_EVENT_FORM_EVENT_TIME}
								</Label>
								<Input
									className="Form-input"
									type="time"
									name="eventTime"
									id="eventTime"
									value={formData.eventTime}
									onChange={handleChange}
									required
								/>
							</FormGroup>
						</Col>
					</Row>
					<Row>
						<Col md={6}>
							<FormGroup>
								<Label htmlFor="city">{staticMsg.EDIT_EVENT_FORM_CITY}</Label>
								<Input
									className="Form-input"
									placeholder="city"
									name="city"
									id="city"
									value={formData.city}
									onChange={handleChange}
								/>
							</FormGroup>
						</Col>
						<Col md={2}>
							<FormGroup>
								<Label htmlFor="state" />
								<Input
									className="Form-input"
									type="select"
									name="state"
									id="state"
									value={formData.state}
									onChange={handleChange}>
									<option>state</option>
									{states.map((state) => <option key={uuid()}>{state}</option>)}
								</Input>
							</FormGroup>
						</Col>
						<Col md={4}>
							<FormGroup>
								<Label htmlFor="country" />
								<Input
									type="select"
									id="country"
									name="country"
									value={formData.country}
									className="Form-input"
									onChange={handleChange}>
									<option>country</option>
									{countries.map((country) => (
										<option key={uuid()}>{country}</option>
									))}
								</Input>
							</FormGroup>
						</Col>
					</Row>
					<FormGroup>
						<Label htmlFor="imgUrl" />
						<Input
							id="imgUrl"
							name="imgUrl"
							className="Form-input"
							value={formData.imgUrl}
							onChange={handleChange}
						/>
					</FormGroup>
					{formErrors &&
					showAlert && (
						<Alert
							type="danger"
							messages={[
								"There was an error updating your event, please try again."
							]}
						/>
					)}

					<Button className="Form-btn btn-success">Save</Button>
				</Form>
			</div>
		</div>
	);
};

export default EditEventForm;
