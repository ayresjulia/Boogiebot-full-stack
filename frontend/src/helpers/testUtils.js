import React from "react";
import UserContext from "./UserContext";
import EventContext from "./EventContext";

const demoUser = {
	username: "testuser",
	firstName: "testfirst",
	lastName: "testlast",
	email: "test@test.net",
	imgUrl: null
};

const testEvents = [
	{
		id: 1,
		title: "Anniversary",
		description: "Anniversary",
		eventDate: "2021-05-19",
		eventTime: "13:00",
		city: "Minsk",
		state: "CA",
		country: "Belarus",
		imgUrl:
			"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
		hostUsername: "testuser"
	},
	{
		id: 2,
		title: "Wedding",
		description: "Wedding",
		eventDate: "2021-05-19",
		eventTime: "13:00",
		city: "Minsk",
		state: "CA",
		country: "Belarus",
		imgUrl:
			"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
		hostUsername: "testuser"
	}
];

const UserProvider = ({ children, currentUser = demoUser }) => (
	<UserContext.Provider value={{ currentUser }}>{children}</UserContext.Provider>
);

const EventProvider = ({ children, events = testEvents }) => (
	<EventContext.Provider value={{ events }}>{children}</EventContext.Provider>
);

export { UserProvider, EventProvider };
