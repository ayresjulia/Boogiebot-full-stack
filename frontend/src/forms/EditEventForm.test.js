import { render } from "@testing-library/react";
import React from "react";
import EditEventForm from "./EditEventForm";
import { useParams, MemoryRouter } from "react-router-dom";
import { EventProvider } from "../helpers/testUtils";

const userInfo = {
	username: "testuser",
	isAdmin: false
};
describe("<EditEventForm.js />", () => {
	it("can tell mocked from unmocked functions", () => {
		expect(jest.isMockFunction(useParams)).toBe(true);
		expect(jest.isMockFunction(MemoryRouter)).toBe(false);
	});
});

it("renders without crashing", async () => {
	useParams.mockReturnValue({ id: 1 });
	render(
		<MemoryRouter>
			<EventProvider>
				<EditEventForm currentUser={userInfo} />
			</EventProvider>
		</MemoryRouter>
	);
});

it("matches snapshot", async () => {
	useParams.mockReturnValue({ id: 1 });
	const { asFragment } = render(
		<MemoryRouter>
			<EventProvider>
				<EditEventForm currentUser={userInfo} />
			</EventProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a field in the form", async () => {
	useParams.mockReturnValue({ id: 1 });
	const { getByText } = render(
		<MemoryRouter>
			<EventProvider>
				<EditEventForm currentUser={userInfo} />
			</EventProvider>
		</MemoryRouter>
	);
	expect(getByText("Edit Anniversary")).toBeInTheDocument();
});
