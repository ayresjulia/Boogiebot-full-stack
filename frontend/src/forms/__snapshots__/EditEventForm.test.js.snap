// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form"
  >
    <div
      class="Form-left"
    >
      <span
        class="Form-update"
      >
        tip.
      </span>
      <p>
        don't forget about younger audience, set up a cool kids table at your party
      </p>
    </div>
    <div
      class="Form-right"
    >
      <div
        class="font-weight-bold text-center mb-card-title event-title"
      >
        Edit Anniversary
      </div>
      <form
        class="Form-body"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="title"
              >
                Title
              </label>
              <input
                class="Form-input form-control"
                id="title"
                name="title"
                type="text"
                value="Anniversary"
              />
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="description"
              >
                Description
              </label>
              <input
                class="Form-input form-control"
                id="description"
                name="description"
                type="text"
                value="Anniversary"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="eventDate"
              >
                Event Date
              </label>
              <input
                class="Form-input form-control"
                id="eventDate"
                name="eventDate"
                required=""
                type="date"
                value="2021-05-19"
              />
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="eventTime"
              >
                Event Time
              </label>
              <input
                class="Form-input form-control"
                id="eventTime"
                name="eventTime"
                required=""
                type="time"
                value="13:00"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="city"
              >
                City
              </label>
              <input
                class="Form-input form-control"
                id="city"
                name="city"
                placeholder="city"
                type="text"
                value="Minsk"
              />
            </div>
          </div>
          <div
            class="col-md-2"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="state"
              />
              <select
                class="Form-input form-control"
                id="state"
                name="state"
              >
                <option>
                  state
                </option>
                <option>
                  AL
                </option>
                <option>
                  AK
                </option>
                <option>
                  AS
                </option>
                <option>
                  AZ
                </option>
                <option>
                  AR
                </option>
                <option>
                  CA
                </option>
                <option>
                  CO
                </option>
                <option>
                  CT
                </option>
                <option>
                  DE
                </option>
                <option>
                  DC
                </option>
                <option>
                  FM
                </option>
                <option>
                  FL
                </option>
                <option>
                  GA
                </option>
                <option>
                  GU
                </option>
                <option>
                  HI
                </option>
                <option>
                  ID
                </option>
                <option>
                  IL
                </option>
                <option>
                  IN
                </option>
                <option>
                  IA
                </option>
                <option>
                  KS
                </option>
                <option>
                  KY
                </option>
                <option>
                  LA
                </option>
                <option>
                  ME
                </option>
                <option>
                  MH
                </option>
                <option>
                  MD
                </option>
                <option>
                  MA
                </option>
                <option>
                  MI
                </option>
                <option>
                  MN
                </option>
                <option>
                  MS
                </option>
                <option>
                  MO
                </option>
                <option>
                  MT
                </option>
                <option>
                  NE
                </option>
                <option>
                  NV
                </option>
                <option>
                  NH
                </option>
                <option>
                  NJ
                </option>
                <option>
                  NM
                </option>
                <option>
                  NY
                </option>
                <option>
                  NC
                </option>
                <option>
                  ND
                </option>
                <option>
                  MP
                </option>
                <option>
                  OH
                </option>
                <option>
                  OK
                </option>
                <option>
                  OR
                </option>
                <option>
                  PW
                </option>
                <option>
                  PA
                </option>
                <option>
                  PR
                </option>
                <option>
                  RI
                </option>
                <option>
                  SC
                </option>
                <option>
                  SD
                </option>
                <option>
                  TN
                </option>
                <option>
                  TX
                </option>
                <option>
                  UT
                </option>
                <option>
                  VT
                </option>
                <option>
                  VI
                </option>
                <option>
                  VA
                </option>
                <option>
                  WA
                </option>
                <option>
                  WV
                </option>
                <option>
                  WI
                </option>
                <option>
                  WY
                </option>
              </select>
            </div>
          </div>
          <div
            class="col-md-4"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="country"
              />
              <select
                class="Form-input form-control"
                id="country"
                name="country"
              >
                <option>
                  country
                </option>
                <option>
                  Afghanistan
                </option>
                <option>
                  Albania
                </option>
                <option>
                  Algeria
                </option>
                <option>
                  Andorra
                </option>
                <option>
                  Angola
                </option>
                <option>
                  Anguilla
                </option>
                <option>
                  Antigua & Barbuda
                </option>
                <option>
                  Argentina
                </option>
                <option>
                  Armenia
                </option>
                <option>
                  Aruba
                </option>
                <option>
                  Australia
                </option>
                <option>
                  Austria
                </option>
                <option>
                  Azerbaijan
                </option>
                <option>
                  Bahamas
                </option>
                <option>
                  Bahrain
                </option>
                <option>
                  Bangladesh
                </option>
                <option>
                  Barbados
                </option>
                <option>
                  Belarus
                </option>
                <option>
                  Belgium
                </option>
                <option>
                  Belize
                </option>
                <option>
                  Benin
                </option>
                <option>
                  Bermuda
                </option>
                <option>
                  Bhutan
                </option>
                <option>
                  Bolivia
                </option>
                <option>
                  Bosnia & Herzegovina
                </option>
                <option>
                  Botswana
                </option>
                <option>
                  Brazil
                </option>
                <option>
                  British Virgin Islands
                </option>
                <option>
                  Brunei
                </option>
                <option>
                  Bulgaria
                </option>
                <option>
                  Burkina Faso
                </option>
                <option>
                  Burundi
                </option>
                <option>
                  Cambodia
                </option>
                <option>
                  Cameroon
                </option>
                <option>
                  Canada
                </option>
                <option>
                  Cape Verde
                </option>
                <option>
                  Cayman Islands
                </option>
                <option>
                  Chad
                </option>
                <option>
                  Chile
                </option>
                <option>
                  China
                </option>
                <option>
                  Colombia
                </option>
                <option>
                  Congo
                </option>
                <option>
                  Cook Islands
                </option>
                <option>
                  Costa Rica
                </option>
                <option>
                  Cote D'Ivoire
                </option>
                <option>
                  Croatia
                </option>
                <option>
                  Cruise Ship
                </option>
                <option>
                  Cuba
                </option>
                <option>
                  Cyprus
                </option>
                <option>
                  Czech Republic
                </option>
                <option>
                  Denmark
                </option>
                <option>
                  Djibouti
                </option>
                <option>
                  Dominica
                </option>
                <option>
                  Dominican Republic
                </option>
                <option>
                  Ecuador
                </option>
                <option>
                  Egypt
                </option>
                <option>
                  El Salvador
                </option>
                <option>
                  Equatorial Guinea
                </option>
                <option>
                  Estonia
                </option>
                <option>
                  Ethiopia
                </option>
                <option>
                  Falkland Islands
                </option>
                <option>
                  Faroe Islands
                </option>
                <option>
                  Fiji
                </option>
                <option>
                  Finland
                </option>
                <option>
                  France
                </option>
                <option>
                  French Polynesia
                </option>
                <option>
                  French West Indies
                </option>
                <option>
                  Gabon
                </option>
                <option>
                  Gambia
                </option>
                <option>
                  Georgia
                </option>
                <option>
                  Germany
                </option>
                <option>
                  Ghana
                </option>
                <option>
                  Gibraltar
                </option>
                <option>
                  Greece
                </option>
                <option>
                  Greenland
                </option>
                <option>
                  Grenada
                </option>
                <option>
                  Guam
                </option>
                <option>
                  Guatemala
                </option>
                <option>
                  Guernsey
                </option>
                <option>
                  Guinea
                </option>
                <option>
                  Guinea Bissau
                </option>
                <option>
                  Guyana
                </option>
                <option>
                  Haiti
                </option>
                <option>
                  Honduras
                </option>
                <option>
                  Hong Kong
                </option>
                <option>
                  Hungary
                </option>
                <option>
                  Iceland
                </option>
                <option>
                  India
                </option>
                <option>
                  Indonesia
                </option>
                <option>
                  Iran
                </option>
                <option>
                  Iraq
                </option>
                <option>
                  Ireland
                </option>
                <option>
                  Isle of Man
                </option>
                <option>
                  Israel
                </option>
                <option>
                  Italy
                </option>
                <option>
                  Jamaica
                </option>
                <option>
                  Japan
                </option>
                <option>
                  Jersey
                </option>
                <option>
                  Jordan
                </option>
                <option>
                  Kazakhstan
                </option>
                <option>
                  Kenya
                </option>
                <option>
                  Kuwait
                </option>
                <option>
                  Kyrgyz Republic
                </option>
                <option>
                  Laos
                </option>
                <option>
                  Latvia
                </option>
                <option>
                  Lebanon
                </option>
                <option>
                  Lesotho
                </option>
                <option>
                  Liberia
                </option>
                <option>
                  Libya
                </option>
                <option>
                  Liechtenstein
                </option>
                <option>
                  Lithuania
                </option>
                <option>
                  Luxembourg
                </option>
                <option>
                  Macau
                </option>
                <option>
                  Macedonia
                </option>
                <option>
                  Madagascar
                </option>
                <option>
                  Malawi
                </option>
                <option>
                  Malaysia
                </option>
                <option>
                  Maldives
                </option>
                <option>
                  Mali
                </option>
                <option>
                  Malta
                </option>
                <option>
                  Mauritania
                </option>
                <option>
                  Mauritius
                </option>
                <option>
                  Mexico
                </option>
                <option>
                  Moldova
                </option>
                <option>
                  Monaco
                </option>
                <option>
                  Mongolia
                </option>
                <option>
                  Montenegro
                </option>
                <option>
                  Montserrat
                </option>
                <option>
                  Morocco
                </option>
                <option>
                  Mozambique
                </option>
                <option>
                  Namibia
                </option>
                <option>
                  Nepal
                </option>
                <option>
                  Netherlands
                </option>
                <option>
                  Netherlands Antilles
                </option>
                <option>
                  New Caledonia
                </option>
                <option>
                  New Zealand
                </option>
                <option>
                  Nicaragua
                </option>
                <option>
                  Niger
                </option>
                <option>
                  Nigeria
                </option>
                <option>
                  Norway
                </option>
                <option>
                  Oman
                </option>
                <option>
                  Pakistan
                </option>
                <option>
                  Palestine
                </option>
                <option>
                  Panama
                </option>
                <option>
                  Papua New Guinea
                </option>
                <option>
                  Paraguay
                </option>
                <option>
                  Peru
                </option>
                <option>
                  Philippines
                </option>
                <option>
                  Poland
                </option>
                <option>
                  Portugal
                </option>
                <option>
                  Puerto Rico
                </option>
                <option>
                  Qatar
                </option>
                <option>
                  Reunion
                </option>
                <option>
                  Romania
                </option>
                <option>
                  Russia
                </option>
                <option>
                  Rwanda
                </option>
                <option>
                  Saint Pierre & Miquelon
                </option>
                <option>
                  Samoa
                </option>
                <option>
                  San Marino
                </option>
                <option>
                  Satellite
                </option>
                <option>
                  Saudi Arabia
                </option>
                <option>
                  Senegal
                </option>
                <option>
                  Serbia
                </option>
                <option>
                  Seychelles
                </option>
                <option>
                  Sierra Leone
                </option>
                <option>
                  Singapore
                </option>
                <option>
                  Slovakia
                </option>
                <option>
                  Slovenia
                </option>
                <option>
                  South Africa
                </option>
                <option>
                  South Korea
                </option>
                <option>
                  Spain
                </option>
                <option>
                  Sri Lanka
                </option>
                <option>
                  St Kitts & Nevis
                </option>
                <option>
                  St Lucia
                </option>
                <option>
                  St Vincent
                </option>
                <option>
                  St. Lucia
                </option>
                <option>
                  Sudan
                </option>
                <option>
                  Suriname
                </option>
                <option>
                  Swaziland
                </option>
                <option>
                  Sweden
                </option>
                <option>
                  Switzerland
                </option>
                <option>
                  Syria
                </option>
                <option>
                  Taiwan
                </option>
                <option>
                  Tajikistan
                </option>
                <option>
                  Tanzania
                </option>
                <option>
                  Thailand
                </option>
                <option>
                  Timor L'Este
                </option>
                <option>
                  Togo
                </option>
                <option>
                  Tonga
                </option>
                <option>
                  Trinidad & Tobago
                </option>
                <option>
                  Tunisia
                </option>
                <option>
                  Turkey
                </option>
                <option>
                  Turkmenistan
                </option>
                <option>
                  Turks & Caicos
                </option>
                <option>
                  Uganda
                </option>
                <option>
                  Ukraine
                </option>
                <option>
                  United Arab Emirates
                </option>
                <option>
                  United Kingdom
                </option>
                <option>
                  United States
                </option>
                <option>
                  Uruguay
                </option>
                <option>
                  Uzbekistan
                </option>
                <option>
                  Venezuela
                </option>
                <option>
                  Vietnam
                </option>
                <option>
                  Virgin Islands (US)
                </option>
                <option>
                  Yemen
                </option>
                <option>
                  Zambia
                </option>
                <option>
                  Zimbabwe
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="imgUrl"
          />
          <input
            class="Form-input form-control"
            id="imgUrl"
            name="imgUrl"
            type="text"
            value=""
          />
        </div>
        <button
          class="Form-btn btn-success btn btn-secondary"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
